server/controllers/usersController.js

```diff
@@ -181,10 +191,22 @@ class UsersController {
       const payload = user;
       const token = await generateToken(payload);
 
+      const profile = {
+        email: user.email,
+        isConfirmed: user.isConfirmed,
+        createdAt: user.createdAt,
+        deletedAt: user.deletedAt,
+        firstname: user['Profile.firstname'],
+        lastname: user['Profile.lastname'],
+        bio: user['Profile.bio'],
+        image: user['Profile.image'],
+        updatedAt: user['Profile.updatedAt'],
+        username: user['Profile.username'],
+      };
       // respond with token
       return response
         .status(200)
-        .json({ token, id: user.id });
+        .json({ token, id: user.id, profile });
```

Comment:
#### Critical
- Include it inside the user object encrypted inside the token rather.
- Or we could have a fetch profile route that is fetched after successful login and put inside our redux store so that we don't tamper with this.

Reply:
We would add a new endpoint that accepts the authentication token in the header and retrieves a user profile on successful authentication.
GET /api/v1/profiles